### PR TITLE
Add bb code

### PR DESCRIPTION
**Pull request by itkg-nanne**

### CODE REVIEW

File: DisplayBundle/BBcode/InternalLinkDefinition.php [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+            $parameterBag = new ParameterBag($parameters);
+
+            $query = array();
+            if (array_key_exists('query', $parameters)) {
+                $query = array($parameters['query']);
+            }
+
+            $uri = '#';
+            foreach ($this->urlGenerator->all() as $router) {
```
Comment (nicolasThal) [inline DisplayBundle/BBcode/InternalLinkDefinition.php]: as said today, I think that we could directly create a router that will be able to generate this kind of route

File: DisplayBundle/BBcode/InternalLinkDefinition.php [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+    /**
+     * @param BBcodeElementNodeInterface $el
+     * @param bool                       $preview
+     */
+    protected function generateHtml(BBcodeElementNodeInterface $el, $preview = false)
+    {
+        $children = $el->getChildren();
+        $option = $el->getAttribute();
+        if ($preview) {
```
Comment (nicolasThal) [inline DisplayBundle/BBcode/InternalLinkDefinition.php]: you could suppress the if and add only one return

Comment (itkg-nanne) [inline DisplayBundle/BBcode/InternalLinkDefinition.php]: I don't think so. generateHtml with preview = true is used in backoffice from getPreviewHtml method (and is not seeing front routers) and preview = false in front with routers from getHtml @nicolasThal

File: DisplayBundle/Resources/views/BBcode/link.html.twig [hunk 3]
```diff
@@ -0,0 +1 @@
+<a href="{{ href }}" data-options="{{ options }}">{{ text }}</a>
```
Comment (Thiblef) [inline DisplayBundle/Resources/views/BBcode/link.html.twig]: eof

File: DisplayBundle/BBcode/InternalLinkDefinition.php [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+namespace OpenOrchestra\DisplayBundle\BBcode;
+
+use OpenOrchestra\BBcodeBundle\Definition\BBcodeDefinition;
+use OpenOrchestra\BBcodeBundle\ElementNode\BBcodeElementNodeInterface;
+use OpenOrchestra\BBcodeBundle\ElementNode\BBcodeElementNode;
+use Symfony\Component\Templating\EngineInterface;
+use OpenOrchestra\DisplayBundle\Manager\NodeManager;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+
```
Comment (Thiblef) [inline DisplayBundle/BBcode/InternalLinkDefinition.php]: remove useless spaces

File: DisplayBundle/BBcode/InternalLinkDefinition.php (hunk 2 above)
Comment (nicolasThal) [inline DisplayBundle/BBcode/InternalLinkDefinition.php]: I would refactor like that :

```

            $parameters = array(
                    'href' => '#',
                    'options' => html_entity_decode($option['link']),
                    'text' => $children[0]->getAsBBCode(),
                )
            );
    if (!$preview) {
            $parameters = json_decode(html_entity_decode($option['link']), true);
            if (!array_key_exists('aliasId', $parameters)) {
                $parameters['aliasId'] = 0;
            }
            $uri = '#';
            try {
                $query = $parameters['query'];
                $linkName = $this->nodeManager->getNodeRouteNameWithParameters($parameters);
                unset($parameters['id']);
                unset($parameters['site']);
                unset($parameters['query']);
                try {
                    $uri = $this->urlGenerator->generate($linkName, $parameters, UrlGeneratorInterface::ABSOLUTE_PATH).$query;
                } catch(RouteNotFoundException $e) {
                }
            } catch (NodeNotFoundException $e) {
            }
            
        }

return $this->templating->render(
                'OpenOrchestraDisplayBundle::BBcode/link.html.twig',
                array(
                    'href' => $uri,
                    'options' => '',
                    'text' => $children[0]->getAsBBCode(),
                )
            );

**Comment by itkg-nanne** [comment]
done @nicolasThal , @Thiblef

**Pull request merged**